### PR TITLE
Remove backwards compatible workaround in SnapshotSerializer?

### PR DESCRIPTION
**Pull request by ortibazar**
Refs  #16197

### CODE REVIEW

File: akka-persistence/src/main/scala/akka/persistence/serialization/SnapshotSerializer.scala [hunk 1]
```diff
@@ -56,24 +50,46 @@ class SnapshotSerializer(val system: ExtendedActorSystem) extends BaseSerializer
...
+
+    out.toByteArray
+  }
+
+  private def headerFromBinary(bytes: Array[Byte]): (Int, String) = {
+    val in = new ByteArrayInputStream(bytes)
+    val serializerId = readInt(in)
+
+    if ((serializerId & 0xEDAC) == 0xEDAC) // Java Serialization magic value
+      throw new NotSerializableException(s"Replaying events from akka 2.3.x version is not supported or do not use serializerId[$serializerId].")
```
Comment (ortigali) [inline akka-persistence/src/main/scala/akka/persistence/serialization/SnapshotSerializer.scala]: Could you offer suitable message?

**Comment by drewhk** [comment]
OK TO TEST

**Comment by johanandren** [comment]
PLS BUILD

File: akka-persistence/src/main/scala/akka/persistence/serialization/SnapshotSerializer.scala (hunk 1 above)
Comment (patriknw) [inline akka-persistence/src/main/scala/akka/persistence/serialization/SnapshotSerializer.scala]: "Replaying snapshot from akka 2.3.x version is not supported any more"

File: akka-persistence-shared/src/test/scala/akka/persistence/serialization/SerializerSpec.scala [hunk 2]
```diff
@@ -108,13 +108,13 @@ class SnapshotSerializerPersistenceSpec extends AkkaSpec(customSerializers) {
...
-      val deserializedDataStr = new String(deserialized.data.asInstanceOf[Array[Byte]], UTF_8)
-      dataStr should ===(deserializedDataStr)
+      val cause = intercept[NotSerializableException] {
+        serializer.fromBinary(bytes, None).asInstanceOf[Snapshot]
+      }
+      cause.getMessage should startWith("Replaying events from akka 2.3.x")
     }
 
-    "be able to read snapshot created with akka 2.3.6 and Scala 2.11" in {
+    "throw error when reads snapshot created with akka 2.3.6 and Scala 2.11" in {
```
Comment (patriknw) [inline akka-persistence-shared/src/test/scala/akka/persistence/serialization/SerializerSpec.scala]: Could you add a similar test that verifies that latest code can read a snapshot created with 2.4.17 (or the commit before this PR). Just to be sure that the changes (readInt/writeInt) didn't break anything.

**Review by patriknw — APPROVED** [review]
LGTM, thanks

**Review by johanandren — APPROVED** [review]
LGTM, thanks!

**Pull request merged**